skimage/_shared/filters.py

```diff
@@ -134,8 +134,8 @@ def gaussian(
     image = convert_to_float(image, preserve_range)
     float_dtype = _supported_float_type(image.dtype)
     image = image.astype(float_dtype, copy=False)
-    if (output is not None) and (not np.issubdtype(output.dtype, np.floating)):
-        raise ValueError("Provided output data type is not float")
+    if (out is not None) and (not np.issubdtype(out.dtype, np.floating)):
+        raise ValueError(f"dtype of `out` must be a floating type, was {out.dtype!r}")
```

Comment:
Sorry, iterating here:
```suggestion
        raise ValueError(f"dtype of `out` must be float; provided {out.dtype!r}.")
```
Either "`out` should be of floating-point dtype" or "dtype of `out` should be float" (right?).

Reply:
Addressed in https://github.com/scikit-image/scikit-image/pull/7225/commits/f01f9085e261e1989fe453a23d5526a3a495260c. "I felt like "got" conveyed the same meaning as "provided" but is a bit less advanced vocabulary."